### PR TITLE
Promotion from AMD internal branch for 2023.Q3.1

### PR DESCRIPTION
**Pull request by WenqingLiAMD**

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 2729132629eaff83641cbf6d7fae284e9cbaf71a
<details><summary>CTS tests (Failed: 0/211876)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35146/70631 (49.8%)</li>
<li>Failed: 0/70631 (0.0%)</li>
<li>Not Supported: 35485/70631 (50.2%)</li><li>Warnings: 0/70631 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39023/70629 (55.3%)</li>
<li>Failed: 0/70629 (0.0%)</li>
<li>Not Supported: 31606/70629 (44.7%)</li><li>Warnings: 0/70629 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39057/70616 (55.3%)</li>
<li>Failed: 0/70616 (0.0%)</li>
<li>Not Supported: 31559/70616 (44.7%)</li><li>Warnings: 0/70616 (0.0%)</li>
</ul></details>


</ul></details>

File: include/vkgcDefs.h [hunk 1]
```diff
@@ -83,12 +83,16 @@
...
 //  | -------- | ----------------------------------------------------------------------------------------------------- |
-//  |     63.0 | Add enableColorExportShader to GraphicsPipelineBuildInfo.                                             |
+//  |     64.0 | Add enableColorExportShader to GraphicsPipelineBuildInfo.                                             |
+//  |     63.0 | Add Atomic Counter, its default descriptor and map its concretType to Buffer.                         |
+//  |     62.1 | Add ApiXfbOutData GraphicsPipelineBuildInfo                                                           |
 //  |     62.0 | Default to the compiler getting the GPURT library directly, and move shader library info into RtState |
-//  |     61.14| Add rasterStream to rsState                                                                           |
+//  |     61.16| Add replaceSetWithResourceType to PipelineOptions                                                     |
+//  |     61.15| Add disableReadFirstLaneWorkaround to PipelineShaderOptions                                           |
+//  |     61.14| Add rasterStream to rsState
```
Comment (nhaehnle) [inline include/vkgcDefs.h:92]: Missing | at the end of the line. Perhaps that needs to be added internally.

**Review by nhaehnle — COMMENTED** [review]
We really need to get that typo checker set up internally...

Anyway, this mostly seems fine, I did notice one formatting issue inline.

File: llpc/test/shaderdb/gfx11/SgprUserDataInit_Fs.pipe [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+; CHECK-NEXT:         .threads_per_subgroup: 0x100
+; CHECK-NEXT:       .gs_vgpr_comp_cnt: 0x1
+; CHECK-NEXT:       .ia_multi_vgt_param_piped:
+; CHECK-NEXT:         .primgroup_size: 0x7f
+; CHECK-NEXT:       .max_verts_per_subgroup: 0x80
+; CHECK-NEXT:       .pa_cl_clip_cntl:
+; CHECK-NEXT:         .dx_linear_attr_clip_ena: true
+; CHECK-NEXT:         .rasterization_kill: false
+; CHECK-NEXT:         .vte_vport_provoke_disable: false
+; CHECK-NEXT:       .pa_cl_vs_out_cntl: {}
```
Comment (xuechen417) [inline llpc/test/shaderdb/gfx11/SgprUserDataInit_Fs.pipe:185]: @WenqingLiAMD The failure of SgprUserDataInit_Fs.pipe can be fixed by removing line 185 : `pa_cl_vs_out_cntl: {}`. The fix is made in porting https://github.amd.com/AMD-Radeon-Driver/llpc/pull/2506.

File: llpc/test/shaderdb/gfx11/SgprUserDataInit_Cs.pipe [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+; CHECK-LABEL: amdgpu_cs_main:
+; CHECK:         s_getpc_b64 s[0:1]
+; CHECK-NEXT:    v_mov_b32_e32 v2, 0
+; CHECK-NEXT:    s_mov_b32 s3, s1
+; CHECK-NEXT:    s_load_b256 s[0:7], s[2:3], 0x0
+; CHECK-NEXT:    s_waitcnt lgkmcnt(0)
+; CHECK-NEXT:    buffer_load_b64 v[0:1], off, s[4:7], 0
+; CHECK-NEXT:    s_waitcnt vmcnt(0)
+; CHECK-NEXT:    buffer_store_format_xy v[0:1], v2, s[0:3], 0 idxen
+; CHECK-NEXT:    s_sendmsg sendmsg(MSG_DEALLOC_VGPRS)
```
Comment (xuechen417) [inline llpc/test/shaderdb/gfx11/SgprUserDataInit_Cs.pipe:146]: @WenqingLiAMD https://github.amd.com/AMD-Radeon-Driver/llpc/commit/a0dfcc18473848b47d9fe0a0a2b78d9ddadc30bd will insert s_nop before s_sendmsg sendmsg(MSG_DEALLOC_VGPRS)

**Comment by amdvlk-admin** [comment]
### Test summary for commit 44faffff4dab25bec4c6a21a5c971ef4e0f76cf4
<details><summary>CTS tests (Failed: 0/211879)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35146/70631 (49.8%)</li>
<li>Failed: 0/70631 (0.0%)</li>
<li>Not Supported: 35485/70631 (50.2%)</li><li>Warnings: 0/70631 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39025/70632 (55.3%)</li>
<li>Failed: 0/70632 (0.0%)</li>
<li>Not Supported: 31607/70632 (44.7%)</li><li>Warnings: 0/70632 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39057/70616 (55.3%)</li>
<li>Failed: 0/70616 (0.0%)</li>
<li>Not Supported: 31559/70616 (44.7%)</li><li>Warnings: 0/70616 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit ec8742633b98926982a966dbc5ec93a8dd097a9d
<details><summary>CTS tests (Failed: 0/211879)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35146/70631 (49.8%)</li>
<li>Failed: 0/70631 (0.0%)</li>
<li>Not Supported: 35485/70631 (50.2%)</li><li>Warnings: 0/70631 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39025/70632 (55.3%)</li>
<li>Failed: 0/70632 (0.0%)</li>
<li>Not Supported: 31607/70632 (44.7%)</li><li>Warnings: 0/70632 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39057/70616 (55.3%)</li>
<li>Failed: 0/70616 (0.0%)</li>
<li>Not Supported: 31559/70616 (44.7%)</li><li>Warnings: 0/70616 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 5f1d711ed0dcdfced2c1b8fba05e26a45e4f4c02
<details><summary>CTS tests (Failed: 0/211879)</summary><ul><li>Built with version 1.3.3.1</li>
<details open><summary>Rhel 9.0, Gfx10</summary><ul>

<li>Passed: 35146/70631 (49.8%)</li>
<li>Failed: 0/70631 (0.0%)</li>
<li>Not Supported: 35485/70631 (50.2%)</li><li>Warnings: 0/70631 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 39025/70632 (55.3%)</li>
<li>Failed: 0/70632 (0.0%)</li>
<li>Not Supported: 31607/70632 (44.7%)</li><li>Warnings: 0/70632 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Navi2x</summary><ul>

<li>Passed: 39057/70616 (55.3%)</li>
<li>Failed: 0/70616 (0.0%)</li>
<li>Not Supported: 31559/70616 (44.7%)</li><li>Warnings: 0/70616 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request closed without merging**